ynr/apps/candidates/tests/test_models.py

```diff
@@ -104,3 +149,87 @@ def test_has_results_and_no_results(self):
         all_winners = set(has_results + has_winner_no_result)
         self.assertEqual(set(Ballot.objects.has_results()), all_winners)
         self.assertEqual(set(Ballot.objects.no_results()), set(no_results))
+
+
+class TestBallotQuerySetMethods(
+    TestCase, BallotsWithResultsMixin, SingleBallotStatesMixin
+):
+    def create_memberships(self, ballot, parties, candidates_per_party=1):
+        for i in range(candidates_per_party):
+            for party in parties:
+                person = PersonFactory()
+                MembershipFactory(ballot=ballot, person=person, party=party)
+
+    def test_uncontested_no_candidates(self):
+        election = self.create_election("Strensall Local Election")
+        post = self.create_post(post_label="Strensall")
+        ballot = self.create_ballot(
+            ballot_paper_id="local.york.strensall.2019-05-02",
+            election=election,
+            post=post,
+            winner_count=1,
+            candidates_locked=True,
+        )
+        uncontested_ballots = Ballot.objects.uncontested()
+        self.assertIn(ballot, uncontested_ballots)
+
+    def test_uncontested_unlocked(self):
+        election = self.create_election("Strensall Local Election")
+        post = self.create_post(post_label="Strensall")
+        ballot = self.create_ballot(
+            ballot_paper_id="local.york.strensall.2020-05-06",
+            election=election,
+            post=post,
+            winner_count=1,
+            candidates_locked=False,
+        )
+        parties = self.create_parties(1)
+        memberships = self.create_memberships(ballot, parties)
```

Comment:
This is a very minor thing, but something to bear in mind generally in future. There is no need to assign this action to the `memberships` variable here. For two reasons actually:
1. the `self.create_memberships` method doesn't actually return anything, so this is always going to be `None`
2. You don't later use the `memberships` variable anywhere in the test - this is the more general thing to remember. Only assign variables if you then intend to use it in some way. Otherwise, simply call the method without assigning a variable:

```
parties = self.create_parties(1)
self.create_memberships(ballot, parties)
```

(this comment is more just about keeping the code clean and to best practices rather than anything practical as the tests will still pass either way)

Reply:
100%, makes sense.